IntelPresentMon/CommonUtilities/Qpc.cpp

```diff
@@ -0,0 +1,65 @@
+#include "Qpc.h"
+#include "win/WinAPI.h"
+#include "log/Log.h"
+#include <thread>
+
+
+namespace pmon::util
+{
+	uint64_t GetCurrentTimestamp() noexcept
+	{
+		LARGE_INTEGER timestamp;
+		if (!QueryPerformanceCounter(&timestamp)) {
+			pmlog_error("qpc failed").hr().every(50);
+		}
+		return (uint64_t)timestamp.QuadPart;
+	}
+	double GetTimestampPeriodSeconds() noexcept
+	{
+		LARGE_INTEGER freq;
+		if (!QueryPerformanceFrequency(&freq)) {
+			pmlog_error("qpc frequency failed").hr().every(5);
+		}
+		return 1.0 / double(freq.QuadPart);
+	}
+	void SpinWaitUntilTimestamp(uint64_t timestamp) noexcept
+	{
+		while (GetCurrentTimestamp() < timestamp) {
+			std::this_thread::yield();
+		}
+	}
+	double TimestampDeltaToSeconds(uint64_t start, uint64_t end, double period) noexcept
```

Comment:
Are you always guaranteed to have end be greater than start?

Reply:
Yes, but it would be a good idea to foolproof this for future usage / changes!